Replace 'Remove from Reusable blocks' with 'Manage Reusable blocks'.

wallstead opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/master/docs/contributors/repository-management.md#pull-requests. -->

## Description
<!-- Please describe what you have changed or added -->
* Removed ReusableBlockDeleteButton component and added ReusableBlocksManageButton in its place.
* On click, ReusableBlocksManageButton sends the user to the reusable blocks management screen.
* Added '@wordpress/url' dependency to the package. This is used to build the  url for the reusable blocks management screen.

Fixes #12791.

## How has this been tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
* Clicked the button when there were no changes in the post, successfully sent to reusable blocks management screen.
* Clicked the button when there were changes in the post, a warning is shown saying there are unsaved changes.
* Ran `npm test`.

## Screenshots <!-- if applicable -->

![image](https://user-images.githubusercontent.com/9662992/99345868-28ab3f80-2848-11eb-8520-c241aca2e65e.png)

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->
Replaces the 'Remove from Reusable blocks' button in block settings menu with a 'Manage Reusable blocks' button that takes the user to the reusable blocks management screen.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

wallstead committed:
Replace 'Remove from Reusable blocks' with 'Manage Reusable blocks'.

* Removed ReusableBlockDeleteButton component and added ReusableBlocksManageButton in its place.

* On click, ReusableBlocksManageButton sends the user to the reusable blocks management screen.

* Added '@wordpress/url' dependency to the package. This is used to build the  url for the reusable blocks management screen.

Fixes #12791.

talldan commented on packages/reusable-blocks/package.json:
```diff
@@ -35,6 +35,7 @@
 		"@wordpress/i18n": "file:../i18n",
 		"@wordpress/icons": "file:../icons",
 		"@wordpress/notices": "file:../notices",
+		"@wordpress/url": "^2.19.0",
```
This dependency should be declared as a file link the same as the other `@wordpress/` dependencies above.

talldan commented on packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-blocks-manage-button.js:
```diff
@@ -0,0 +1,55 @@
+		( select ) => {
+			const { getBlock } = select( 'core/block-editor' );
+			const { canUser } = select( 'core' );
+			const blockObj = getBlock( clientId );
```
Would be great to avoid an abbreviation here (the coding standards says to use full names where possible - https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#naming-conventions).

I think I'd go with naming this `reusableBlock` and change the later variable to `reusableBlockPost` (if it's needed).

talldan commented on packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-blocks-manage-button.js:
```diff
@@ -0,0 +1,55 @@
+			const { canUser } = select( 'core' );
+			const blockObj = getBlock( clientId );
+
+			const reusableBlock =
+				blockObj && isReusableBlock( blockObj )
+					? select( 'core' ).getEntityRecord(
+							'postType',
+							'wp_block',
+							blockObj.attributes.ref
+					  )
+					: null;
```
When I tested, `blockObj.attributes.ref` and `reusableBlock.id` were the same value, so just wondering if this call to `getEntityRecord` is needed? Is this to check that a post actually exists for the editor that's about to be loaded?

Worth noting that while the reusable block is saving I believe it has a temporary id (mentioned here - https://github.com/WordPress/gutenberg/issues/19115), so we might want to check for that.

wallstead replied:
Good point. I don't think `getEntityRecord` is needed here. It was a remnant from the `ReusableBlockDeleteButton` component, but I don't think it is relevant for the user experience in this case. I removed it in my most recent commit.

As for the temporary id, I cannot replicate the ref attribute being anything other than a number id. I looked for documentation on this for a while but came up empty-handed.

talldan commented on packages/reusable-blocks/src/components/reusable-blocks-menu-items/reusable-blocks-manage-button.js:
```diff
@@ -0,0 +1,55 @@
+		<BlockSettingsMenuControls>
+			<MenuItem
+				onClick={ () => {
+					window.location = addQueryArgs( 'edit.php', {
```
I think this menu item should ideally render as a link (an `<a>` element instead of a `<button>`). 

`MenuItem` can take the same props as `Button`, so it should be possible to pass the value to the `href` prop:
https://github.com/WordPress/gutenberg/blob/master/packages/components/src/button/index.js#L23

I do wonder if it should open in a new tab 🤔

wallstead replied:
Good to know this! I used your suggestion in my most recent commit. 

Regarding the new tab functionality, I'm totally open to changing it but for now I'm leaving the functionality consistent with the "Manage all reusable blocks" button shown below. It doesn't open a new tab.

![image](https://user-images.githubusercontent.com/9662992/99609494-dfccc580-29c4-11eb-8455-a5276f23bb37.png)

talldan reviewed (commented):
Thanks so much for working on this @wallstead, the code looks really solid for your first contribution. I left a few comments, but nothing too complex.

wallstead committed:
Fix logic and dependency for new 'Manage Reusable blocks' menu item.

* Use file link for new @wordpress/url dependency.

* Rename abbreviated blockObj variable to reusableBlock to follow coding standards.

* Simplified logic to check if the 'Manage Reusable blocks' menu item should be visible.

* Use MenuItem's href prop to create an a tag instead of relying on onClick on a button.

Fixes #27026.

talldan reviewed (approved):
This is looking good! Thanks for making those few extra changes.

The only remaining thing I noticed is that the capitalization isn't consistent with the 'Manage all reusable blocks' option in the editor menu. Then I looked closer, and lots of strings seem inconsistent, so I made an issue (https://github.com/WordPress/gutenberg/issues/27085). 

I think 'Manage all reusable blocks' is the incorrect label here, so lets leave it as is.

If you can resolve the merge conflict I'd be happy to merge. Congrats on your first contribution!

wallstead committed:
Update package lock. (#27026)

wallstead committed:
Remove unused e2e test for deleting reusable blocks. (#27026)

wallstead commented:
@talldan I fixed the merge conflict and updated the package-lock to fix a failing check. Also, I saw a left-over e2e test for the "Remove from Reusable blocks" button we are replacing. I removed that unused test to fix another failing check. Hopefully, that's okay.

talldan commented:
Nice work! Would be great to add an e2e test for the new 'Manage reusable blocks' option if you feel like taking a crack at it 😄
